### PR TITLE
These are the documentation changes for PR #81.

### PR DESCRIPTION
**Pull request by schweitzpgi**
For some unknown reason they cause the sphinx builder script to crash and burn.

### CODE REVIEW

**Comment by schweitzpgi** [comment]
@bettinaheim These documentation changes look fairly mundane, but the bots don't like them. I can't figure out why. Can you take a look?

**Comment by bettinaheim** [comment]
> @bettinaheim These documentation changes look fairly mundane, but the bots don't like them. I can't figure out why. Can you take a look?

@schweitzpgi Sure. FYI, in case you haven't seen it, if you go to the [CI run](https://github.com/NVIDIA/cuda-quantum/actions/runs/4693766090?pr=86), you will see a `build` folder in the artifacts when the docs generation fails. You can click on it to download it. On one hand that folder contains the log files that are mentioned by the docs generation script, and on the other hand, that folder is what contains the build to reproduce the issue locally.

For this PR, the log shows the following sphinx error:
```
/workspaces/cuda-quantum/docs/sphinx/specification/cudaq/synthesis.rst:61: ERROR: Error in "code-block" directive:
maximum 1 argument(s) allowed, 5 supplied.

.. code-block:: cpp
   cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
```

You are lacking an empty line after code-block. See also https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block

File: docs/sphinx/specification/cudaq/operations.rst [hunk 1]
```diff
@@ -99,4 +99,8 @@ qubit state is a :code:`|0>` (positive polarity) or :code:`|1>` (negative polari
 By default all control qubits are assumed to convey positive polarity. 
 The syntax for negating the polarity is the not operator preceeding the
 control qubit (e.g., :code:`x<cudaq::ctrl>(!q, r)`, 
-for :code:`cudaq::qubits` :code:`q` and :code:`r`).
\ No newline at end of file
+for :code:`cudaq::qubits` :code:`q` and :code:`r`). Negating the polarity of
+control qubits is supported in :code:`swap` and the gates in sets :code:`NAME`
```
Comment (bettinaheim) [inline docs/sphinx/specification/cudaq/operations.rst:103]: What does NAME and ROTATION_NAME refer to? I'll check once the docs build completes, but I think that will show up in the docs in that form.

File: docs/sphinx/specification/cudaq/synthesis.rst [hunk 2]
```diff
@@ -54,6 +54,13 @@ The :code:`cudaq::control(...)` function takes as input an instantiated pure
...
-using any pertinent synthesis strategy available. The :code:`cudaq::adjoint(...)`
-function takes as input an instantiated pure device quantum kernel
-(or specified template type) and the remaining arguments for the kernel. 
\ No newline at end of file
+using any pertinent synthesis strategy available. Qubits may be aggregated into
+a range of control qubits with or without the use of the :code:`operator!`
+`negated polarity operator <https://nvidia.github.io/cuda-quantum/specification/cudaq/operations.html>`_.
+
+.. code-block:: cpp
+   cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
```
Comment (bettinaheim) [inline docs/sphinx/specification/cudaq/synthesis.rst:62]:
```suggestion

   cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
```

**Comment by schweitzpgi** [comment]
> > @bettinaheim These documentation changes look fairly mundane, but the bots don't like them. I can't figure out why. Can you take a look?
> 
> @schweitzpgi Sure. FYI, in case you haven't seen it, if you go to the [CI run](https://github.com/NVIDIA/cuda-quantum/actions/runs/4693766090?pr=86), you will see a `build` folder in the artifacts when the docs generation fails. You can click on it to download it. On one hand that folder contains the log files that are mentioned by the docs generation script, and on the other hand, that folder is what contains the build to reproduce the issue locally.
> 
> For this PR, the log shows the following sphinx error:
> 
> ```
> /workspaces/cuda-quantum/docs/sphinx/specification/cudaq/synthesis.rst:61: ERROR: Error in "code-block" directive:
> maximum 1 argument(s) allowed, 5 supplied.
> 
> .. code-block:: cpp
>    cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
> ```
> 
> You are lacking an empty line after code-block. See also https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block

I did download the zip file, but I found nothing that led me to the problem inside.

File: docs/sphinx/specification/cudaq/operations.rst (hunk 1 above)
Comment (schweitzpgi) [reply docs/sphinx/specification/cudaq/operations.rst:103]: See lines 88 and 89.

**Review by bettinaheim — APPROVED** [review]

**Comment by bettinaheim** [comment]
> > > @bettinaheim These documentation changes look fairly mundane, but the bots don't like them. I can't figure out why. Can you take a look?
> > 
> > 
> > @schweitzpgi Sure. FYI, in case you haven't seen it, if you go to the [CI run](https://github.com/NVIDIA/cuda-quantum/actions/runs/4693766090?pr=86), you will see a `build` folder in the artifacts when the docs generation fails. You can click on it to download it. On one hand that folder contains the log files that are mentioned by the docs generation script, and on the other hand, that folder is what contains the build to reproduce the issue locally.
> > For this PR, the log shows the following sphinx error:
> > ```
> > /workspaces/cuda-quantum/docs/sphinx/specification/cudaq/synthesis.rst:61: ERROR: Error in "code-block" directive:
> > maximum 1 argument(s) allowed, 5 supplied.
> > 
> > .. code-block:: cpp
> >    cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
> > ```
> >
> > You are lacking an empty line after code-block. See also https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block
> 
> I did download the zip file, but I found nothing that led me to the problem inside.

If you look at the failing step, it tells you 
```
Sphinx exit code: 1
Documentation generation failed with exit code 12.
Check the logs in /workspaces/cuda-quantum/build/logs, and the documentation build output in /workspaces/cuda-quantum/build/docs
```
The build folder dumped as artifact is this /workspaces/cuda-quantum/build folder. The error message I posted above is in the logs/sphinx_error.txt file.

**Pull request merged**